src/ui/profile/ProfileSelection.js

```diff
@@ -0,0 +1,170 @@
+// eslint-disable-next-line no-redeclare
+/* global chrome */
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import styles from '../styles/profile.module.css';
+import AddProfile from './AddProfile';
+import BeyondLoader from './BeyondLoader';
+import Profiles from './Profile';
+
+const LoadedProfile = (profileModel) => ({
+  status: 'loaded',
+  profile: profileModel,
+});
+
+const ErrorProfile = (error, profile) => ({
+  status: 'error',
+  error,
+  profile,
+});
+
+const LoadingProfile = (profile) => ({
+  status: 'loading',
+  profile,
+});
+
+function ProfileSelection(props) {
+  const { onCharacterReady } = props;
+  const [profiles, setProfiles] = useState(new Map());
```

Comment:
You did an excellent job here making sure to create a new `Map` every time you update the state. Most React newbies that I've seen forget that state needs to be kept immutable.

However, given that things need to be kept immutable, a `Map` is not a particularly optimal choice. What I would do instead is use an object, especially considering that you're keying by a primitive value:

```js
const [profiles, setProfiles] = useState({});
```

I'll leave comments where this would change your code below.

Reply:
I want the data structure to be ordered because I will eventually support dragging profiles around to re-organize them

Reply:
That makes sense, but a `Map` is not an ordered collection. The entries may happen to be ordered based on order of insertion, but I don't believe that you can always count on that.

If ordering is what you want then I'd recommend just using an array (similar to how you store it in local storage), which is obviously guaranteed to be ordered. I don't believe you'd expect the list of profiles to be very large, so linear time lookups are not going to have a significant impact.